camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelEndpointAnnotator.java

```diff
@@ -91,10 +91,13 @@ private void extractSetValue(EndpointValidationResult result, Set<String> valida
             for (String entry : validationSet) {
                 String propertyValue = entry;
 
-                int propertyIdx = fromElement.indexOf(propertyValue);
+                int startIdxQueryParameters = fromElement.indexOf("?" + propertyValue);
+                startIdxQueryParameters = (startIdxQueryParameters == -1) ? fromElement.indexOf("&" + propertyValue) : fromElement.indexOf("?");
```

Comment:
I wonder if the endpoint do not have any query parameters, eg without a `?`. Would that not return `-1` and maybe cause the index to be off by one or something?